pkg/providers/instancetype/suite_test.go

```diff
@@ -654,6 +654,37 @@ var _ = Describe("InstanceTypes", func() {
 		}
 		Expect(nodes.Len()).To(Equal(1))
 	})
+	It("should launch instances w/ instance storage for ephemeral storage resource requests when disks are mounted for ephemeral-storage", func() {
+		nodeClass.Spec.AMIFamily = lo.ToPtr(v1beta1.AMIFamilyAL2)
+		nodeClass.Spec.InstanceStorePolicy = lo.ToPtr(v1beta1.InstanceStorePolicyRAID0)
+		ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+		pod := coretest.UnschedulablePod(coretest.PodOptions{
+			ResourceRequirements: v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceEphemeralStorage: resource.MustParse("5000Gi")},
+			},
+		})
+		ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod)
+		ExpectScheduled(ctx, env.Client, pod)
```

Comment:
Worth checking which instance type that this scheduled to and the size of the ephemeral storage on that instance type? Same as the launch template testing?

Reply:
added in https://github.com/aws/karpenter-provider-aws/pull/4735/commits/4bf56c65efa9007b3aee0871e6d554e370fa549b